src/process.py

```diff
@@ -0,0 +1,84 @@
+import os
+import shlex
+import signal
+import subprocess
+import tempfile
+
+#Process(f'xdotool search --all --name {name}').wait().stdout
+
+# with Process(CMD, capture=False):
+        
+#     p = Process('roslaunch test test.launch')
+#     p.wait()
+        
+class Process:
+    """
+    Convenience wrapper for subprocess.Popen. Allows to:
+    - pass cmd as a string despite not using shell
+    - append env variables to spawned process context
+    - capture output (stdout, stderr),
+    - send SIGINT to spawned process group to stop it
+    """
+
+    def __init__(self, cmd, shell=False, env=None, capture=True):
+        self.cmd = cmd
+        self._rc = None
+        self._files = {}
+        kw = dict(preexec_fn=os.setsid, shell=shell, close_fds=True)
```

Comment:
This results in error on Windows because `os.setsid` is platform dependent.

I do not know whether it is applicable solution, I just omit this parameter if it is not Linux:

```python
kw = dict(
    preexec_fn=os.setsid if sys.platform.startswith('linux') else None,
    shell=shell,
    close_fds=True,
)
```

And import `sys` on top.

Reply:
I just removed it, it is not really needed. Thank you! I am also afraid a little bit about the kill() -> it is using os.killpg() but let see.